packages/foam-vscode/src/features/folder-rename-handler.ts

```diff
@@ -0,0 +1,1112 @@
+              // If it's a simple identifier (no path separators), and it would still be unique, preserve it
+              let shouldPreserveOriginal = false;
+
+              // Only consider preserving simple identifiers (no path separators)
+              // Path-based identifiers like "features/daily-notes" need to be updated when folders are renamed
+              if (
+                !originalTarget.includes('/') &&
+                FoamWorkspace.isIdentifier(originalTarget)
+              ) {
+                // This is a simple identifier like [[note]] or [[daily-notes]]
+                // Check if it would still resolve uniquely to our target after the rename
+
+                // Get all resources that match this identifier (workspace is already updated)
+                const candidatesAfterRename =
+                  this.foam.workspace.listByIdentifier(originalTarget);
+
+                // If there's exactly one candidate and it's our target, the simple identifier remains unambiguous
+                if (
+                  candidatesAfterRename.length === 1 &&
+                  candidatesAfterRename[0].uri.isEqual(targetFoamUri)
+                ) {
+                  shouldPreserveOriginal = true;
+                  Logger.debug(
+                    `  → Preserving simple wikilink "${originalTarget}" as it remains unambiguous after rename`
+                  );
+                } else {
+                  Logger.debug(
+                    `  → Simple wikilink "${originalTarget}" has ${candidatesAfterRename.length} candidates, will update to avoid ambiguity`
+                  );
+                }
+              } else {
+                Logger.debug(
+                  `  → Path-based wikilink "${originalTarget}" will be updated for folder rename`
+                );
+              }
+
+              if (shouldPreserveOriginal) {
+                // Don't update the wikilink - it's still valid
+                Logger.debug(
+                  `  → Wikilink in "${vscode.workspace.asRelativePath(
+                    actualSourceUri
+                  )}" preserved: "${connection.link.rawText}"`
+                );
+              } else {
+                // getIdentifier will produce the correct string for inside [[...]]
+                // respecting defaultExtension and uniqueness
+                let newLinkContent: string;
+                if (originalTarget.includes('/')) {
+                  const wsFolder =
+                    vscode.workspace.getWorkspaceFolder(newFileUri);
+                  if (wsFolder) {
+                    const rel = path
+                      .relative(wsFolder.uri.fsPath, newFileUri.fsPath)
+                      .replace(/\\/g, '/');
+                    newLinkContent = rel.replace(path.extname(rel), '');
+                  } else {
+                    newLinkContent =
+                      this.foam.workspace.getIdentifier(targetFoamUri);
+                  }
```

Comment:
I wonder if there is a way where you can work at the level of the `FoamWorkspace` instead of having to manipulate paths and extensions. 
E.g. if (this is off the top of my head, not thought through) you "clone" the Foam workspace object, update the resource objects in there (basically, simulating the renaming was completed), and the using `getIdentifier` to retrieve the new identifier

The good thing about that approach is that it should be pretty standard, but the devil might be in the details (including performance)

What do you think?